cherab/tools/observers/bolometry.py

```diff
@@ -687,6 +695,350 @@ def etendue_single_run(_):
+
+        basis_x = basis_x.normalise()
+        basis_y = basis_y.normalise()
+        normal_vec = basis_x.cross(basis_y)
+        self._slit = slit
+        self._curvature_radius = curvature_radius
+        self._accumulate = accumulate
+
+        # setup root bolometer foil transform
+        translation = translate(centre_point.x, centre_point.y, centre_point.z)
+        rotation = rotate_basis(normal_vec, basis_y)
+
+        super().__init__([slit.target], pixels=pixels, width=width,
+                         targetted_path_prob=1.0, parent=parent, pipelines=[],
+                         transform=translation * rotation, name=detector_id)
+        self.pixel_samples = 1000
+        self.spectral_bins = 1
+        self.quiet = True
+
+        # Update pipeline based on units
+        self.units = units
+
+        # round off the detector corners, if applicable
+        if self._curvature_radius > 0:
+            mask_corners(self)
+
+    def __repr__(self):
+        """Returns a string representation of this BolometerIRVB object."""
+        return "<BolometerIRVB - " + self.name + ">"
+
+    @property
+    def pixels_as_foils(self):
+        XAXIS = Vector3D(1, 0, 0)
+        YAXIS = Vector3D(0, 1, 0)
+        nx, ny = self.pixels
+        pixel_width = self.width / nx
+        pixel_height = self.height / ny
+        # Foil pixels are defined in the foil's local coordinate system
+        foil_bottom_left = Point3D(-self.width / 2, -self.height / 2, 0)
+        pixels = []
+        for x in range(nx):
+            pixel_column = []
+            for y in range(ny):
+                pixel_centre = (foil_bottom_left
+                                + (x + 0.5) * XAXIS * pixel_width
+                                + (y + 0.5) * YAXIS * pixel_height)
+                pixel = BolometerFoil(
+                    detector_id="IRVB pixel ({},{})".format(x + 1, y + 1),
+                    centre_point=pixel_centre, basis_x=XAXIS, dx=pixel_width,
+                    basis_y=YAXIS, dy=pixel_height, slit=self._slit,
+                    units=self.units, accumulate=False, parent=self
+                )
+                pixel_column.append(pixel)
+            pixels.append(pixel_column)
+        return np.asarray(pixels, dtype='object')
+
+    @property
+    def height(self):
+        return self.width * self.pixels[1] / self.pixels[0]
+
+    @property
+    def centre_point(self):
+        return Point3D(0, 0, 0).transform(self.to_root())
+
+    @property
+    def normal_vector(self):
+        return Vector3D(0, 0, 1).transform(self.to_root())
+
+    @property
+    def basis_x(self):
+        return Vector3D(1, 0, 0).transform(self.to_root())
+
+    @property
+    def basis_y(self):
+        return Vector3D(0, 1, 0).transform(self.to_root())
+
+    @property
+    def sightline_vectors(self):
+        return np.asarray(
+            [[pixel.centre_point.vector_to(self._slit.centre_point) for pixel in pixel_column]
```

Comment:
Re previous comment on using a general affine transform. The centre should ideal be at the origin of the new space i.e `Point3D(0, 0, 0)`

Reply:
This will have to wait until the `BolometerFoil` api is fixed, I think. Other methods which treat the IRVB as a collection of foils, such as `trace_sightlines` and `as_sightlines`, use the `BolometerFoil` conventions as that's what the IRVB pixels are broken down into for these methods.